platform/linux-generic/odp_packet.c

```diff
@@ -1910,6 +1911,47 @@ int _odp_packet_copy_md_to_packet(odp_packet_t srcpkt, odp_packet_t dstpkt)
 	return dst_uarea_size < src_uarea_size;
 }
 
+static uint32_t packet_sum_ones_comp16_32(odp_packet_hdr_t *pkt_hdr,
+					  uint32_t offset,
+					  uint32_t len,
+					  odp_bool_t odd_offset)
+{
+	uint32_t sum = 0;
+
+	if (offset + len > pkt_hdr->frame_len)
+		return 0;
+
+	while (len > 0) {
+		uint32_t seglen = 0; /* GCC */
```

Comment:
this is not because of gcc. Gcc warns because packet_map() can return seglen unmodified and code will make no sense here.

Reply:
Not sure if this is still the case. The original code had issues with GCC only (not clang). `seglen` is always set as long as the returned addr isn't `NULL` and `seglen` itself isn't `NULL` indicating that the return is not needed.